### PR TITLE
External api integration

### PR DESCRIPTION
**Pull request by alexdavcudriz**
## Summary
- Feign configuration was added.
- Client models was created.
- Client delegates was created.
- Isbn search was added.
## Trello Card
https://trello.com/c/oktaL917/22-integraci%C3%B3n-api-externa

### CODE REVIEW

File: src/main/java/wolox/training/client/delegate/OpenLibraryDelegate.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+        HashMap<String, Object> bookInfoResponse = openLibraryFeignClient
+                .findBookByIsbn(new BooksParam(isbnParam.toString()));
+
+        if (!bookInfoResponse.isEmpty()) {
+            return Optional.ofNullable(objectMapper
+                    .convertValue(bookInfoResponse.get(isbnParam.toString()), BookInfoDto.class));
+        }
+
+        return Optional.empty();
```
Comment (facucompe) [inline src/main/java/wolox/training/client/delegate/OpenLibraryDelegate.java]: I don't like the Optional ofNullable inside a if

Comment (facucompe) [reply src/main/java/wolox/training/client/delegate/OpenLibraryDelegate.java]: Maybe this works as well:
```
return Optional.ofNullable(bookInfoResponse.get(isbnParam.toString()))
    .map(bookInfo -> objectMapper.convertValue(bookInfo, BookInfoDto.class));
```

File: src/main/java/wolox/training/services/BookService.java [hunk 2]
```diff
@@ -37,4 +52,19 @@ public void delete(Long id) {
...
+        Optional<Book> bookOptional = bookRepository.findByIsbn(isbn);
+        if (bookOptional.isPresent()) {
+            return Collections.singleton(bookOptional.get());
+        }
+
+        Book bookFound = openLibraryDelegate.findBookByIsbn(isbn)
+                .map(bookInfoDto -> bookMapper.bookInfoDtoToToEntity(isbn, bookInfoDto))
+                .orElseThrow(BookNotFoundException::new);
+
+        return Collections.singleton(bookRepository.save(bookFound));
```
Comment (facucompe) [inline src/main/java/wolox/training/services/BookService.java]: Same here. If you use optional try not asking if it's present.

Comment (facucompe) [reply src/main/java/wolox/training/services/BookService.java]:
```
return bookRepository.findByIsbn(isbn)
     .map(book -> Collections.singleton(book))
     .orElse(openLibraryBook(isbn));
```

**Review by facucompe — CHANGES_REQUESTED** [review]
What about Service test? We will see web mocks later to test OpenLibrary Component Test

**Review by facucompe — APPROVED** [review]

**Review by jdhiguita — APPROVED** [review]

**Pull request merged**